### ISSUE TITLE
No status in `diffSummary` with  `--name-status` 

### ISSUE THREAD

**Issue by gonardfreeman** [steveukx/git-js#953]
diffSummary command with `--name-status` option doesn't return status of the file (A,M,D etc).
Here code line where you can see that it is [not included into output](https://github.com/steveukx/git-js/blob/8a31c45a1359248cb38d768d22a5e29903ceef8a/simple-git/src/lib/parsers/parse-diff-summary.ts#L88C27-L88C27)

also status isn't [included into type](https://github.com/steveukx/git-js/blob/8a31c45a1359248cb38d768d22a5e29903ceef8a/simple-git/typings/response.d.ts#L138)

I'm suggesting such changes:
[Here](https://github.com/steveukx/git-js/blob/8a31c45a1359248cb38d768d22a5e29903ceef8a/simple-git/typings/response.d.ts#L153)
```typescript
export interface DiffResultTextFile {
   file: string;
   changes: number;
   insertions: number;
   deletions: number;
   binary: false;
   status: string | null
}
```
[And here](https://github.com/steveukx/git-js/blob/8a31c45a1359248cb38d768d22a5e29903ceef8a/simple-git/src/lib/parsers/parse-diff-summary.ts#L88C27-L88C27)
```typescript
const nameStatusParser = [
   new LineParser<DiffResult>(/([ACDMRTUXB])\s*(.+)$/, (result, [_status, file]) => {
      result.changed++;
      result.files.push({
         file,
         changes: 0,
         insertions: 0,
         deletions: 0,
         binary: false,
         status: _status
      });
   }),
];
```

**Comment by frpanico96** [comment]
This feature would be very appreciated.

A pull request have been opened in order to introduce this feature.

https://github.com/steveukx/git-js/pull/908

It is possible to introduce the feature?

**Comment by steveukx** [comment]
Thank you for the detailed issue.

This functionality is released in version 3.22.0 which is on npm now.

The released version differs from your suggestion in that `status` field is a `DiffNameStatus | undefined` where [DiffNameStatus](https://github.com/steveukx/git-js/blob/main/simple-git/src/lib/tasks/diff-name-status.ts) is the enum of values that can be matched.
